src/main/java/com/commercetools/sync/products/helpers/VariantReferenceResolver.java

```diff
@@ -125,50 +124,35 @@ public VariantReferenceResolver(@Nonnull final ProductSyncOptions productSyncOpt
 
     @Nonnull
     CompletionStage<AttributeDraft> resolveAttributeReference(@Nonnull final AttributeDraft attributeDraft) {
+
         final JsonNode attributeDraftValue = attributeDraft.getValue();
+
         if (attributeDraftValue == null) {
             return CompletableFuture.completedFuture(attributeDraft);
         }
-        if (attributeDraftValue.isArray()) {
-            return resolveAttributeSetReferences(attributeDraft);
-        } else {
-            if (isProductReference(attributeDraftValue)) {
-                return getResolvedIdFromKeyInReference(attributeDraftValue)
-                    .thenApply(productIdOptional ->
-                        productIdOptional.map(productId ->
-                            AttributeDraft.of(attributeDraft.getName(), createProductReferenceJson(productId)))
-                                         .orElse(attributeDraft));
-            }
-            return CompletableFuture.completedFuture(attributeDraft);
-        }
-    }
 
-    @Nonnull
-    private CompletionStage<AttributeDraft> resolveAttributeSetReferences(
-        @Nonnull final AttributeDraft attributeDraft) {
-        final JsonNode attributeDraftValue = attributeDraft.getValue();
-        final Spliterator<JsonNode> attributeReferencesIterator = attributeDraftValue.spliterator();
-
-        final Stream<JsonNode> attributeReferenceStream = StreamSupport.stream(attributeReferencesIterator, false)
-                                                                       .filter(Objects::nonNull)
-                                                                       .filter(reference -> !reference.isNull());
+        final JsonNode clonedValue = attributeDraftValue.deepCopy();
```

Comment:
```suggestion
        final JsonNode cloneOfAttributeDraftValue = attributeDraftValue.deepCopy();
```

Reply:
What do you think about `attributeDraftValueClone`?

Reply:
yes, it is better, but you got the idea 😄

Reply:
91fad10